added tcp_user_timeout

Jitendra-Rajak opened the pull request:
<!--- Please review your changes in preview mode -->
<!--- Provide a general summary of your changes in the Title above -->

#### Describe Bug or Feature
pbs_comm not marking node down when there is job with substate 41.


#### Describe Your Change
Added TCP_USER_TIMEOUT switch alongside the other keepalive switches we have. Currently I have added user_timeout of 50000ms i.e 50 seconds. Increasing the user timeouts allows a TCP connection to survive extended periods without end-to-end connectivity. Decreasing the user timeouts allows applications to "fail fast" if so desired. Otherwise it may take up to 20 minutes with the current system defaults in a normal environment.



#### Attach Test and Valgrind Logs/Output
<!--- Please attach your test log output from running the test you added (or from existing tests that cover your changes) -->
<!--- Don't forget to run Valgrind if appropriate and attach the resulting logs -->
[logs_50sec_timeout.txt](https://github.com/PBSPro/pbspro/files/3493226/logs_50sec_timeout.txt)



<!--- Pull Request Guidelines: [Pull Request Guidelines](https://pbspro.atlassian.net/wiki/spaces/DG/pages/1187348483/Pull+Request+Guidelines) -->

nithinj commented:
@Jitendra-Rajak ,
Can you try changing the value in the spec file and verify whether its getting reflected in the logs?

vstumpf commented on src/lib/Libtpp/tpp_dis.c:
```diff
@@ -893,6 +894,7 @@ set_tpp_config(struct pbs_config *pbs_conf,
+	tpp_conf->tcp_user_timeout = DEFAULT_TCP_USER_TIMEOUT;
 
 	/* if set, read them from environment variable PBS_TCP_KEEPALIVE */
 	if ((s = getenv(PBS_TCP_KEEPALIVE))) {
```
Can you change the comment on line 903 to add tcp_user_timeout to the format.

Jitendra-Rajak commented:
Hey @nithinj, I have tried changing the env variable values and yes it is reflecting in the logs.

Jitendra-Rajak commented:
Hey @vstumpf, I have addressed your comments please take the PR for review once again and if there is any further comments do let me know.

Thanks,
Jitendra
